### PR TITLE
RESTWS-722: Allow a user to request for a password reset via email

### PR DESCRIPTION
**Pull request by wluyima**

### CODE REVIEW

File: omod-2.2/pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	<description>OpenMRS module project for Rest Web Services</description>
 
 	<properties>
-                 <openmrs.version.2.2.0-SNAPSHOT>2.2.0-SNAPSHOT</openmrs.version.2.2.0-SNAPSHOT>
+                 <openmrs.version.2.2.0-SNAPSHOT>2.2.0-passwordreset</openmrs.version.2.2.0-SNAPSHOT>
```
Comment (kaweesi) [inline omod-2.2/pom.xml]: we definitely want to revert the version change except if we wanna maintain this in a separate branch

**Comment by kaweesi** [comment]
i think we need to update this to match changes in master or resolve any possible merge conflicts

File: omod-2.2/pom.xml (hunk 1 above)
Comment (wluyima) [reply omod-2.2/pom.xml]: Definitely before the merge

**Comment by dkayiwa** [comment]
@wluyima do you think this is ready for merge?

**Comment by fanyui** [comment]
Hello @dkayiwa  This is the corresponding webservice pull request for password reset as part of my gsoc project which you merged in core. Has any one tested or reviewed?

**Comment by dkayiwa** [comment]
@fanyui do all the tests pass when you run it?

**Comment by fanyui** [comment]
@dkayiwa  Yes they did when i made the pull request. i had to change the pom so that i could build with respect the changes i made in core then. But since you have merged the one in core. i will have to build it again .

**Comment by dkayiwa** [comment]
@fanyui are you available to do it again now?

**Comment by fanyui** [comment]
@dkayiwa  Sure, I will do it in the next hour

**Comment by dkayiwa** [comment]
Awesome! Am anxiously looking forward to it. 😊

**Comment by fanyui** [comment]
@dkayiwa Sorry it took more than required time. am getting some unexpected errors with my .m2/repository i need to delete it but connection for downloading new one is not strong enough. But i'll do it as soon as it good connection is back.

**Comment by dkayiwa** [comment]
Sorry about that. Looking forward to see it done!

**Comment by dkayiwa** [comment]
@fanyui are you still working on this?

**Comment by fanyui** [comment]
@dkayiwa  I have been having some test failures on this. I tried fixing it but i seem to be having limited knowledge about what is happening. But when i deploy the module on the running instance with the changes i get required result but running the tests it fails. Also because i am trying not to allow the code to send mails each time a user runs test on this module, A little help will be appreciated I will create a talk post for assistance or via irc.

**Comment by dkayiwa** [comment]
Yes creating a new pull request looks fine.

**Comment by fanyui** [comment]
@dkayiwa  This is the pull request i made to the branch from where this one was made https://github.com/openmrs/openmrs-module-webservices.rest/pull/375

**Comment by dkayiwa** [comment]
The new pr also has merge conflicts.

**Comment by dkayiwa** [comment]
Closing because of this: https://github.com/openmrs/openmrs-module-webservices.rest/pull/383

**Pull request closed without merging**